### ISSUE TITLE
Sharing links to Focus, opens google search results

### ISSUE THREAD

**Issue by sv-ohorvath** [mozilla-mobile/focus-android#2308]
Build:
Focus Beta 4.1
Focus store vers. 4.0.2
Klar+GeckoView 4.1

Devices:
Samsung Galaxy S8 (Android 7.0)
HTC Desire 820 (Android 6.0.1)

STR:
1. Open one of the following apps:
Amazon
News republic
Google News
2. Open an article.
3. Tap the share button and select Focus or Klar.

Expected: 
The link opens in the selected browser (same as Add to Nightly works).

Actual result:
The page title and URL are sent to Google search in Focus/ Klar.

**Comment by pocmo** [comment]
Flagged it with bug/feature because it's kind of both:

* If the shared text is a URL then we open it, otherwise we do a search
* I assume Fennec does search for a URL inside the text?

**Comment by sv-ohorvath** [comment]
I've shared an article page entirely. It searches for the article url & title I've shared. I skipped one step in the description - updated.

**Comment by ekager** [comment]
The string that we are getting from the intent (from Amazon) looks like "Anker PowerPort Wireless 5 Stand, 5W Standard Qi-Certified Wireless Charger for iPhone X, iPhone 8 / 8 Plus (AC Adapter Not Included) https://www.amazon.com/dp/B0753Z4PGC/ref=cm_sw_r_other_apa_I8TQAbMTSHNDV". 
We check whether it is a URL or not, and then search or load the page. Not sure if we want to try to parse for a URL?

**Comment by pocmo** [comment]
@brampitoyo What do you think?

**Comment by brampitoyo** [comment]
@ekager @pocmo I think user’s intention when they select Share differs based on the application. A few examples.

1. Share -> Twitter, Facebook == post this as a tweet, post it to my wall, etc.
2. Share -> Pinterest, Pocket, Instapaper == save this link to my collection, reading list, etc.
3. Share -> Chrome, Firefox, Focus == open this link there

In other words, users don’t always think that the share action should do item 1, even though the word “Share” that we like to use heavily implies doing item 1.

TL;DR: we should parse the URL out of whatever text strings the app includes, then open that URL.

**Comment by pocmo** [comment]
Fennec seems to use this:
https://dxr.mozilla.org/mozilla-central/source/mobile/android/services/src/main/java/org/mozilla/gecko/sync/setup/activities/WebURLFinder.java#311

**Comment by ekager** [comment]
Thanks @pocmo that seems to work here too :)
